#' ENCODE Unified GRCh38 Exclusion List (previously "blacklist")
#'
#' The blacklist is a unification of three separate blacklists followed by extensive manual curation generated by Anshul Kundaje, Stanford https://www.encodeproject.org/files/ENCFF356LFX/
#' The three blacklists that were unified are as follows
#' 1. The autogenerated blacklist from https://github.com/Boyle-Lab/Blacklist/blob/master/lists/hg38-blacklist.v2.bed.gz . 
#' The corresponding citation is
#' Amemiya, H.M., Kundaje, A. & Boyle, A.P. The ENCODE Blacklist: Identification of Problematic Regions of the Genome. Sci Rep 9, 9354 (2019). https://doi.org/10.1038/s41598-019-45839-z   
#' 2. The autogenerated peakPass blacklist from https://github.com/ewimberley/peakPass/blob/master/blacklists/hg38/peakPass60Perc_sorted.bed
#' The corresponding citation is
#' Charles E. Wimberley and Steffen Heber.Journal of Computational Biology.Feb 2020.259-268.http://doi.org/10.1089/cmb.2019.0295
#' 3. The blacklist capturing peaks in the nuclear genome that originate from reads from mitochondrial genome taken from https://github.com/caleblareau/mitoblacklist/blob/master/peaks/hg38_peaks.narrowPeak . This blacklist was generated by Caleb Lareau.
#' 
#' 
#' @format a GRanges object
#' 
#' @docType data
#' 
#' @usage data("deny")
#' 
"deny"

#' nullranges DNase hypersensitivity peaks in A549 cell example data
#'
#' An example dataset containing narrowPeak file from ENCODE. Retrieve record with 'object[["AH22505"]]' on Annotation Hub. Analysis script is in 'inst/script/DHSA549Hg38.R'.
#' 
#' @format a GRanges object with metadata score, signal value, p/q value and peak.
#' 
#' @docType data
#' 
#' @usage data("DHSA549Hg38")
#' 
"DHSA549Hg38"

#' Predicted CTCF-motif binding sites
#'
#' Genomic coordinates of predicted CTCF binding sites with motif MA0139.1 (Jaspar), in BED format, with strand orientation (directionality of binding). Human (hg19, hg38) and mouse (mm9, mm10) genomes. The binding sites were detected using the FIMO tool of the MEME suite using default settings. Extra columns include motif name (MA0139.1), score, p-value, q-value, and the motif sequence.
#' 
#' Experimental, to be submitted as an AnnotatiohHub Bioconductor package.
#' 
#' 
#' 
#' @format a GRanges object
#' 
#' @docType data
#' 
#' @usage data("CTCF_hg19")
#' 
#' 
#' @references 
#' Dozmorov MG (2021). _CTCF_. https://github.com/mdozmorov/CTCF/CTCF - R
#' package version 0.99.0, <URL: https://github.com/mdozmorov/CTCF>.
#' 
"CTCF_hg19"

#' 10Kb bins from hg19 with GM12878 metadata annotation features
#'
#' 10Kb bins were tiled across hg19 and annotated with CTCF and DNase site features from GM12878. Feature annotations for each bin include 1) the number of CTCF sites, 2) the CTCF signal strength (from peak calls), 3) the number of DNase sites, 4) the DNase signal strength (from signal tracks), and finally 5) the presence/absence of a loop to any other bin.
#'
#' 
#' @format a GRanges object with covariate metadata
#' 
#' @docType data
#' 
#' @usage data("hg19_10kb_bins")
#' 
#' 
"hg19_10kb_bins"

#' CTCF-bound 10Kb paired genomic interactions
#'
#' 10Kb bins were tiled across hg19 then subset by those which contained CTCF sites. All pairs of CTCF-bound 10Kb bins were generated and annotated with feature overlaps from GM12878. Feature annotations include 1) presence/absence of a loop between bin-pairs, 2) the total CTCF signal from both bin-pairs, 3) the number of CTCF sites from both bin-pairs, 4) the distance between bin-pairs, and finally 4) whether a convergent set of CTCF sites exists between bin-pairs.
#' 
#' 
#' @format a GInteractions object with covariate metadata
#' 
#' @docType data
#' 
#' @usage data("hg19_10kb_ctcfBoundBinPairs")
#' 
#' 
"hg19_10kb_ctcfBoundBinPairs"